test/e2e/cro_test.go

```diff
@@ -0,0 +1,149 @@
+package e2e
+
+import (
+	goctx "context"
+	"fmt"
+	"testing"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	v1 "k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/integr8ly/cloud-resource-operator/pkg/apis"
+	"github.com/integr8ly/cloud-resource-operator/pkg/apis/integreatly/v1alpha1"
+	"github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
+
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	errorUtil "github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	retryInterval = time.Second * 20
+	timeout       = time.Second * 60
+)
+
+func TestCRO(t *testing.T) {
+	redisList := &v1alpha1.Redis{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Redis",
+			APIVersion: "integreatly.org/v1alpha1",
+		},
+	}
+	if err := framework.AddToFrameworkScheme(apis.AddToScheme, redisList); err != nil {
+		t.Fatalf("failed to add Redis custom resource scheme to framework: %v", err)
+	}
+
+	postgresList := &v1alpha1.Postgres{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Postgres",
+			APIVersion: "integreatly.org/v1alpha1",
+		},
+	}
+	if err := framework.AddToFrameworkScheme(apis.AddToScheme, postgresList); err != nil {
+		t.Fatalf("failed to add Postgres custom resource scheme to framework: %v", err)
+	}
+
+	// run subtests
+	t.Run("cro-group", func(t *testing.T) {
+		t.Run("Cluster", CROCluster)
+	})
+}
+
+func postgresTest(t *testing.T, f *framework.Framework, ctx framework.TestCtx) error {
+	namespace, err := ctx.GetNamespace()
+	if err != nil {
+		return errorUtil.Wrapf(err, "could not get namespace")
+	}
+	examplePostgres := &v1alpha1.Postgres{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Postgres",
+			APIVersion: "integreatly.org/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "example-postgres",
+			Namespace: namespace,
+		},
+		Spec: v1alpha1.PostgresSpec{
+			SecretRef: &v1alpha1.SecretRef{
+				Name:      "postgres-credentials",
+				Namespace: namespace,
+			},
+			Tier: "development",
+			Type: "workshop",
+		},
+	}
+	// create postgres resource
+	if err := f.Client.Create(goctx.TODO(), examplePostgres, getCleanupOptions(t)); err != nil {
+		return errorUtil.Wrapf(err, "could not create example Postgres")
+	}
+
+	// wait from postgres deployment
+	if err := e2eutil.WaitForDeployment(t, f.KubeClient, namespace, "example-postgres", 1, retryInterval, timeout); err != nil {
```

Comment:
Do we need to wait for this at all anymore?

Reply:
We don't, but I think for logging it might be better to catch the error here, if there is a problem with the deployment, over a timeout on the secret. What do you think?

Reply:
Going to leave this poll for now, till we sort the reconcile time